test/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.FunctionalTests/ViewCompilationOptionsTest.cs

```diff
@@ -20,27 +20,23 @@ public ViewCompilationOptionsTest(TestFixture fixture)
         [Fact]
         public void Precompilation_PreventsRefAssembliesFromBeingPublished()
         {
-            // Arrange
-            using (var deployer = Fixture.CreateDeployment())
-            {
-                // Act
-                var deploymentResult = deployer.Deploy();
+            // Arrange & Act
+            var deploymentResult = Fixture.CreateDeployment();
 
-                // Assert
-                Assert.False(Directory.Exists(Path.Combine(deploymentResult.ContentRoot, "refs")));
-            }
+            // Assert
+            Assert.False(Directory.Exists(Path.Combine(deploymentResult.ContentRoot, "refs")));
         }
 
         [Fact]
         public void PublishingWithOption_AllowsPublishingRefAssemblies()
         {
             // Arrange
-            Fixture.PrepareForDeployment();
             var deploymentParameters = Fixture.GetDeploymentParameters();
             deploymentParameters.PublishEnvironmentVariables.Add(
                 new KeyValuePair<string, string>("MvcRazorExcludeRefAssembliesFromPublish", "false"));
+            var logger = Fixture.CreateLogger();
 
-            using (var deployer = ApplicationDeployerFactory.Create(deploymentParameters, Fixture.Logger))
+            using (var deployer = ApplicationDeployerFactory.Create(deploymentParameters, logger))
```

Comment:
What about this particular call requires a different `ILogger` instance than `Fixture.Logger` returns? Not clear why `CreateLogger()` needed to change except for this use case.